content/plugins/notes/kubernetes.md

```diff
@@ -0,0 +1,207 @@
+
+        ```
+        Kubernetes master is running at ...
+        ```
+
+    -   `name`
+
+        A name to **represent this cluster**, this must be unique within the `clusters`
+        array. This value will be displayed to the final User in the Service Catalog Kubernetes plugin.
+
+    -   `authProvider`
+
+        Determines how the Kubernetes client authenticates with the Kubernetes
+        cluster.
+
+        **Valid values:**
+
+        -   `serviceAccount` - Use of a Kubernetes [service account](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/) to access the Kubernetes API.  
+            **NOTE:** when this is used the `serviceAccountToken` field should also be set (see bellow).
+
+        -   `google` - Use of a user's Google auth token from the [Google auth plugin](https://backstage.io/docs/auth/) to access the Kubernetes API.
+
+    -   `serviceAccountToken` - _optional_
+
+        The Kubernetes service account token to be used when `authProvider` is set to `serviceAccount`.
+
+### Using RBAC Authorization
+
+The current RBAC permissions required are read-only cluster wide, for the
+following objects:
+
+-   pods
+-   services
+-   configmaps
+-   deployments
+-   replicasets
+-   horizontalpodautoscalers
+-   ingresses
+
+An example of a Role to grant read access to pods:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+    namespace: default
+    name: pod-reader
+rules:
+    - apiGroups: ['']
+      resources: ['pods']
+      verbs: ['get', 'watch', 'list']
+```
+
+## Surfacing your Kubernetes components as part of an entity
+
+There are two ways to surface your Kubernetes components as part of an entity.  
+**NOTE:** The label selector takes precedence over the annotation/service id.
+
+### Common `backstage.io/kubernetes-id` label
+
+1. #### Adding the metadata annotation
+
+The following annotation must be added so that Backstage can detect that an entity has Kubernetes components.
+
+```yaml
+# catalog-info.yaml
+metadata:
+    annotations:
+        'backstage.io/kubernetes-id': <ENTITY_NAME>
+    ...
+```
+
+2. #### Labeling your Kubernetes components
+
+In order for your Kubernetes components to show up in the service catalog as a part
+of an entity, your Kubernetes components themselves can have the following label:
+
+```json
+"metadata": {
+  "labels": {
+    "backstage.io/kubernetes-id": <ENTITY_NAME>
+    ...
+  }
+}
+```
+
+### Label selector query annotation
+
+Via a label selector, the user can identify a set of objects.  
+You can write your own custom label selector query that Backstage will use to
+lookup the objects. You can read
+[Labels and Selectors documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)
+for more info.
+
+```yaml
+# catalog-info.yaml
+annotations:
+    'backstage.io/kubernetes-label-selector': 'app=my-app,component=front-end'
+```
+
+## Example of steps to follow
```

Comment:
I love that you added an example. Anytime I find myself reading documentation, a simple and clear example that I can follow is what helps me the most.